components/board/genericlinux/board.go

```diff
@@ -36,27 +36,33 @@ func RegisterBoard(modelName string, gpioMappings map[int]GPIOBoardMapping) {
 				conf resource.Config,
 				logger golog.Logger,
 			) (board.Board, error) {
-				return newBoard(ctx, conf, gpioMappings, logger)
+				newConf, err := resource.NativeConfig[*Config](conf)
+				if err != nil {
+					return nil, err
+				}
+				return NewBoard(ctx, conf.ResourceName().AsNamed(), newConf, gpioMappings, logger)
```

Comment:
I'd like this to have the same signature as before, you can move the native config logic into the newBoard function and use it in there.

Reply:
customlinux package uses the NewBoard function so I need the config parameter type to be the genericlinux.Config, not resource.Config

Reply:
Thus can be changed back now.